harness/determined/common/experimental/trial.py

```diff
@@ -165,59 +196,84 @@ def select_checkpoint(
             resp = bindings.get_GetCheckpoint(self._session, checkpointUuid=uuid)
             return checkpoint.Checkpoint._from_bindings(resp.checkpoint, self._session)
 
-        def get_one(offset: int) -> bindings.v1GetTrialCheckpointsResponse:
+        order_by = None
+        if latest:
+            sort_by = CheckpointSortBy.SORT_BY_BATCH_NUMBER
+            order_by = CheckpointOrderBy.DESC
+
+        if sort_by:
+            order_by = CheckpointOrderBy.ASC if smaller_is_better else CheckpointOrderBy.DESC
+
+        checkpoints = self.get_checkpoints(sort_by=sort_by, order_by=order_by)
+
+        return checkpoints[0] if checkpoints else []
+
+    def get_checkpoints(
+        self,
+        sort_by: Optional[Union[str, CheckpointSortBy]] = None,
+        order_by: Optional[CheckpointOrderBy] = None,
+    ) -> List[checkpoint.Checkpoint]:
+        """
+        Return a list of :class:`~determined.experimental.Checkpoint` instances for the current
+        trial.
+
+        Either ``sort_by`` and ``order_by`` are both specified or neither are.
+
+        Arguments:
+            sort_by (string, :class:`~determined.experimental.CheckpointSortBy`): Which field to
+                sort by. Strings are assumed to be searcher metric names.
```

Comment:
there's only one searcher metric name though, right?  Do you mean "validation metric names"?